### PR TITLE
fix ipv6 concat port to ip address string

### PR DESCRIPTION
**Pull request by szuecs**
fix https://github.com/zalando/skipper/issues/1997
Signed-off-by: Sandor Szücs <sandor.szuecs@zalando.de>

### CODE REVIEW

**Comment by szuecs** [comment]
:+1:

File: dataclients/kubernetes/ingressdefinitions.go [hunk 1]
```diff
@@ -83,7 +85,11 @@ type endpointList struct {
 }
 
 func formatEndpoint(a *address, p *port, protocol string) string {
-	return fmt.Sprintf("%s://%s:%d", protocol, a.IP, p.Port)
+	ip := net.ParseIP(a.IP)
+	if strings.Count(a.IP, ":") >= 2 { // our naive IPv6 check
+		return fmt.Sprintf("%s://[%s]:%d", protocol, ip.String(), p.Port)
+	}
+	return fmt.Sprintf("%s://%s:%d", protocol, ip.String(), p.Port)
```
Comment (AlexanderYastrebov) [inline dataclients/kubernetes/ingressdefinitions.go:92]: There is https://pkg.go.dev/net#JoinHostPort so maybe it should be something like
```go
return protocol + "://" + net.JoinHostPort(a.IP, strconv.Itoa(p.Port))
```

Comment (szuecs) [reply dataclients/kubernetes/ingressdefinitions.go]: done

**Comment by szuecs** [comment]
:+1:

**Comment by AlexanderYastrebov** [comment]
:+1:

**Pull request merged**